src/main/webapp/script.js

```diff
@@ -1,3 +1,96 @@
+class ParameterInput extends HTMLElement {
+  constructor() {
+    super();
+    this.label = document.createElement('label');
+    this.textArea = document.createElement('textarea');
+    this.button = document.createElement('button');
+    this.container = document.createElement('div');
+
+    this.container.appendChild(this.label);
+    this.container.appendChild(this.textArea);
+    this.container.appendChild(this.button);
+  }
+
+  connectedCallback() {
+    this.name = this.getAttribute('name');
+    this.index = parseInt(this.getAttribute('index'));
+    this.parent = this.name + 's';
+
+    this.textArea.rows = '1';
+    this.button.type = 'button';
+    this.button.innerText = 'Add ' + this.name;
+    this.setIndexAttributes();
+
+    this.appendChild(this.container);
+  }
+
+  setIndexAttributes() {
+    var paramName = this.name.toLowerCase() + this.index;
+    this.id = this.name + this.index;
+
+    this.label.innerText = this.name + ' ' + (this.index + 1);
+    this.label.for = paramName;
+
+    this.textArea.name = paramName;
+
+    this.button.onclick = event => {
+      var newParameter = createParameterInput(this.name, this.index + 1);
+      insertParameterInput(this, newParameter);
+    }
+  }
+
+  get text() {
+    return this.textArea.value;
+  }
+
+  get position() {
+    return this.index;
+  }
+
+  get field() {
+    return this.parent;
+  }
+
+  set text(value) {
+    this.textArea.value = value;
+  }
+
+  set position(value)  {
+    this.index = parseInt(value);
+  }
+
+  set field(value) {
+    this.parent = value;
+  }
+}
+customElements.define('parameter-input', ParameterInput);
+
+function createParameterInput(name, index) {
+  var newParameter = document.createElement('parameter-input');
+  newParameter.setAttribute('name', name);
+  newParameter.setAttribute('index', index);
+  newParameter.setAttribute('id', name + index);
+  return newParameter;
+}
+
+function insertParameterInput(previous, parameterInput) {
+  previous.insertAdjacentElement('afterend', parameterInput);
+  updateIndeces(parameterInput.field, parameterInput.position + 1);
+}
+
+function appendParameterInput(fieldName, parameterInput) {
+  const field = document.getElementById(fieldName);
+  field.appendChild(parameterInput);
+}
+
+function updateIndeces(fieldName, startIndex) {
```

Comment:
spell time

Reply:
Wow now I'm even more embarrased

Reply:
I'll change this in the next PR (delete parameters) to avoid merge conflicts